src/utils/Api.js

```diff
@@ -0,0 +1,93 @@
+class Api {
+  constructor({ baseUrl, headers = {} } = {}) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  // TODO create another method called get user info
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    }).then(this._checkResponse);
+  }
+
+  getAppInfo() {
+    // TODO Call getuserinfo in this array
+    return Promise.all([
+      fetch(`${this._baseUrl}/cards`, {
+        headers: this._headers,
+      }).then(this._checkResponse),
+      this.getUserInfo(),
+    ]);
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    }).then(this._checkResponse);
+  }
+
+  // TODO - implement Post/cards- creating a card, go to index.js to see how we did the edit user information //
+  addCard({ name, link }) {
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify({
+        name,
+        link,
+      }),
+    }).then(this._checkResponse);
+  }
+
+  deleteCard(cardId) {
+    return fetch(`${this._baseUrl}/cards/${cardId}`, {
+      method: "DELETE",
+      headers: this._headers,
+    }).then((res) => {
+      if (!res.ok) {
+        console.error(
+          `Failed to delete card with ID: ${cardId}, Status: ${res.status}`
+        );
+        return Promise.reject(`Error: ${res.status}`);
+      }
+      return res.json();
```

Comment:
`.then(this._checkResponse);` should be used here as well.


You don't need this specific error text. You can add this text in `.catch` in `index.js`

Reply:
My remark hasn't been corrected